registry/virtual.go

```diff
@@ -1,50 +1,455 @@
 package registry
 
 import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
 	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
 	"fmt"
+	"io"
+	"io/ioutil"
 	"net/http"
+	"net/url"
 	"os"
+	"path"
 	"path/filepath"
 
 	"github.com/cozy/cozy-apps-registry/asset"
+
+	"github.com/cozy/cozy-apps-registry/space"
+
 	"github.com/cozy/cozy-apps-registry/base"
 	"github.com/go-kivik/kivik/v3"
 )
 
-// FindAppAttachmentFromOverwrite finds if the app icon was overwritten in the
-// virtual space.
-func FindAppAttachmentFromOverwrite(virtualSpaceName, appSlug, filename string) *Attachment {
-	if filename != "icon" {
+func findOverwrittenVersion(s base.VirtualSpace, version *Version) (*Version, error) {
+	db := s.VersionDB()
+	ctx := context.Background()
+	row := db.Get(ctx, version.ID)
+	var t Version
+	if err := row.ScanDoc(&t); err != nil {
+		if kivik.StatusCode(err) == http.StatusNotFound {
+			return nil, nil
```

Comment:
I really don't like returning nil, nil. I have been bitten too many times where the caller function check that the error is nil and then assume that the other argument is not nil. Maybe we can add a boolean `ok` (`return nil, false, nil`) to say if the version has been found.